pkg/controller/elasticsearch/services/services.go

```diff
@@ -9,27 +9,66 @@ import (
 	"math/rand"
 	"strconv"
 
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
-
 	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/common/defaults"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/label"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/network"
 	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
 	"github.com/elastic/cloud-on-k8s/pkg/utils/stringsutil"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 const (
 	globalServiceSuffix = ".svc"
 )
 
+// TransportServiceName returns the name for the transport service associated to this cluster
+func TransportServiceName(esName string) string {
+	return esv1.TransportService(esName)
+}
+
+// NewTransportService returns the transport service associated with the given cluster.
+// It is used by Elasticsearch nodes to talk to each other.
+func NewTransportService(es esv1.Elasticsearch) *corev1.Service {
+	nsn := k8s.ExtractNamespacedName(&es)
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: es.Namespace,
+			Name:      TransportServiceName(es.Name),
+			Labels:    label.NewLabels(nsn),
+		},
+		Spec: corev1.ServiceSpec{
```

Comment:
Should we allow users to control the spec like we do for the HTTP service? For remoting outside of the k8s cluster users probably want this to be of type `LoadBalancer` for example.

Reply:
It makes sense. That said this PR is already pretty large and is targeting clusters inside a same k8s cluster. I would do that in a dedicated PR, I can create an issue to track this effort.

Reply:
👍